components/script/dom/globalscope.rs

```diff
@@ -3142,15 +3143,16 @@ impl GlobalScope {
                 let gamepad = Gamepad::new(&global, index as u32, name, axis_bounds, button_bounds);
 
                 if let Some(window) = global.downcast::<Window>() {
-                    let gamepad_list = window.Navigator().GetGamepads();
+                    let has_gesture = window.Navigator().has_gamepad_gesture();
+                    if has_gesture {
+                        gamepad.set_exposed(true);
+                    }
+                    let gamepad_list = window.Navigator().gamepads();
                     let gamepad_arr: [DomRoot<Gamepad>; 1] = [gamepad.clone()];
                     gamepad_list.add_if_not_exists(&gamepad_arr);
 
-                    // TODO: 3.4 If navigator.[[hasGamepadGesture]] is true:
-                    // TODO: 3.4.1 Set gamepad.[[exposed]] to true.
-
                     if window.Document().is_fully_active() {
-                        gamepad.update_connected(true);
+                        gamepad.update_connected(true, has_gesture);
```

Comment:
here is `has_gesture ` and when you call function you use `should_notify ` I would expect prams name to be the same

Reply:
agree, arg name should be `has_gesture`.